Automattic/jetpack
# [AI] Better openAi prompt

Depending on #28483 to be merged

Fixes #28550

## Proposed changes:
* Handles more post cases when creating the OpenAI prompt
* Slightly Refactor it
* Only display an error when the post is totally empty
* Started to add unit tests

### Other information:
 [X ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No
## Testing instructions:
* Deploy on WPCOM
* Sandbox your favorite site
* Go to `{YOUR_SITE}/wp-admin/post-new.php`
* Add an AI paragraph block to the empty page and you should see
<img width="613" alt="Screenshot 2023-01-25 at 14 48 23" src="https://user-images.githubusercontent.com/790558/214580320-77905884-9184-4c82-95c0-f3ee0c28d6d6.png">

* Entering a title, some content, a category or a tag should ask you to retry
* You can now prompt for each case and see the different prompts in the network console
* All types of prompts are unit-tested in `extensions/blocks/ai-paragraph/test/edit.js`

## E2E tests

Run
```
 pnpm test-extensions extensions/blocks/ai-paragraph/test/edit.js
```
